i18n: Add explanation for locales and automatic language setting (#10008)
godotengine/godot-docs: tutorials/i18n/internationalizing_games.rst

    the current language, making it ideal for things like multiplayer chat where
    the text language may not match the client's language.

Automatically setting a language
--------------------------------

> It is recommended to default to the user's preferred language which can be obtained via :ref:`OS.get_locale_language() <class_OS_public_method_get_locale_language>`. If your game is not available in that language, it will fallback to the one in **Project settings > General > Internationalization > Locale (Advanced settings)** (if empty `en` for English will be used which is recommended).

Reviewer: ```suggestion
It is recommended to default to the user's preferred language which can be obtained via :ref:`OS.get_locale_language() <class_OS_public_method_get_locale_language>`. If your game is not available in that language, it will fall back to the one in **Project settings > General > Internationalization > Locale (Advanced settings)** (if empty `en` for English will be used which is recommended).
```

Nevertheless letting players change the language in game is recommended for various reasons (e.g. translation quality or player preference).

.. tabs::
 .. code-tab:: gdscript

>     var language = "automatic"

Reviewer: ```suggestion

    var language = "automatic"
```

Thanks for contributing :) You're missing a newline here for the code block to work, I think.

    # Load here language from the user settings file
    if language == "automatic":
       var preferred_language = OS.get_locale_language()
       TranslationServer.set_locale(preferred_language)
    else:
       TranslationServer.set_locale(language)

Locale vs. language
-------------------
A Locale is commonly a combination of a language with a region or country but can also contain more information like scripts.

Examples:

- `en`: English language
- `en_GB`: English in Great Britain / British English

> - `en_US`: Enginsh in the USA / American English

Author: 🙈

    the current language, making it ideal for things like multiplayer chat where
    the text language may not match the client's language.

Automatically setting a language
--------------------------------

> It is recommended to default to the user's preferred language which can be obtained via :ref:`OS.get_locale_language() <class_OS_method_get_locale_language>`. If your game is not available in that language, it will fall back to the one in **Project settings > General > Internationalization > Locale (Advanced settings)** (if empty `en` for English will be used which is recommended).

Reviewer 2: ```suggestion
It is recommended to default to the user's preferred language which can be obtained via :ref:`OS.get_locale_language() <class_OS_method_get_locale_language>`. If your game is not available in that language, it will fall back to the :ref:`Fallback <class_ProjectSettings_property_internationalization/locale/fallback>` in **Project Settings > Internationalization > Locale**, or to ``en`` if empty.
```
Add a link to the project settings. (We started doing this more consistently after you made this PR, I think)

> It is recommended to default to the user's preferred language which can be obtained via :ref:`OS.get_locale_language() <class_OS_method_get_locale_language>`. If your game is not available in that language, it will fall back to the :ref:`Fallback <class_ProjectSettings_property_internationalization/locale/fallback>` in **Project Settings > Internationalization > Locale**, or to ``en`` if empty.

Reviewer 3: ```suggestion
It is recommended to default to the user's preferred language which can be obtained via :ref:`OS.get_locale_language() <class_OS_method_get_locale_language>`.
If your game is not available in that language, it will fall back to the :ref:`Fallback <class_ProjectSettings_property_internationalization/locale/fallback>`
in **Project Settings > Internationalization > Locale**, or to ``en`` if empty.
```

Nevertheless letting players change the language in game is recommended for various reasons (e.g. translation quality or player preference).

.. tabs::
 .. code-tab:: gdscript

    var language = "automatic"
    # Load here language from the user settings file
    if language == "automatic":
       var preferred_language = OS.get_locale_language()
       TranslationServer.set_locale(preferred_language)
    else:
       TranslationServer.set_locale(language)

Locale vs. language
-------------------

> A Locale is commonly a combination of a language with a region or country but can also contain more information like scripts.

Reviewer 2: ```suggestion
A :ref:`locale <doc_locales>` is commonly a combination of a language with a region or country, but can also contain information like a script or a variant.
```
No need to capitalize locale here
Slightly clarified the sentence as well
Linked to the list of locales, too

Examples:

> - `en`: English language
> - `en_GB`: English in Great Britain / British English
> - `en_US`: English in the USA / American English
> - `en_DE`: English in Germany

Reviewer 2: ```suggestion
- ``en``: English language
- ``en_GB``: English in Great Britain / British English
- ``en_US``: English in the USA / American English
- ``en_DE``: English in Germany
```
If you meant these to be `code` and not *italics*, you need two backticks for RST
